sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceStrategy.scala

```diff
@@ -253,6 +260,16 @@ class FindDataSourceTable(sparkSession: SparkSession) extends Rule[LogicalPlan]
 
     case UnresolvedCatalogRelation(tableMeta) =>
       DDLUtils.readHiveTable(tableMeta)
+
+    case AppendData(DataSourceV2Relation(t: DataSourceV1Table, output, _), query, false) =>
+      InsertIntoDataSourceCommand(fallbackToV1Relation(t, output), query, overwrite = false)
+
+    case OverwriteByExpression(
+        DataSourceV2Relation(t: DataSourceV1Table, output, _), Literal(true, _), query, false) =>
```

Comment:
The Data Source V1 API don't support static partition overwrite (e.g. `INSERT OVERWRITE t PARTITION(a=1) SELECT ...`). We must fully migrate JDBC to Data Source V2 before supporting it.

Here we only need to match the OverwriteByExpression(..., deleteExpr=Literal(true)), as this is the only thing DS v1 supports.

Reply:
maybe also add a `@TODO` here?